controllers/storagecluster/provider_server.go

```diff
@@ -62,10 +62,16 @@ func (o *ocsProviderServer) ensureDeleted(r *StorageClusterReconciler, instance
 		return reconcile.Result{}, err
 	}
 
+	serviceType := corev1.ServiceTypeClusterIP
+	if instance.Spec.ProviderAPIServerServiceType != nil {
+		serviceType = *instance.Spec.ProviderAPIServerServiceType
+	} else if instance.Spec.HostNetwork {
+		serviceType = corev1.ServiceTypeNodePort
+	}
 	var finalErr error
 
 	for _, resource := range []client.Object{
-		GetProviderAPIServerService(instance),
+		GetProviderAPIServerService(instance.Namespace, serviceType),
```

Comment:
Why do we need to get a full service spec just to delete something? can't we just use the name and namespace?

Reply:
changed.